src/qforte/ucc/spqe.py

```diff
@@ -309,59 +342,17 @@ def get_residual_vector(self, trial_amps):
             # occ => i,j,k,...
             # vir => a,b,c,...
             # sq_op is 1.0(a^ b^ i j) - 1.0(j^ i^ b a)
-            temp_idx = sq_op.terms()[0][2][-1]
-            if temp_idx < int(sum(self._ref)/2): # if temp_idx is an occupied idx
-                sq_creators = sq_op.terms()[0][1]
-                sq_annihilators = sq_op.terms()[0][2]
-            else:
-                sq_creators = sq_op.terms()[0][2]
-                sq_annihilators = sq_op.terms()[0][1]
-
-            destroyed = False
-            denom = 1.0
-
-            basis_I = qforte.QubitBasis(self._nqb)
-            for k, occ in enumerate(self._ref):
-                basis_I.set_bit(k, occ)
-
-            # loop over anihilators
-            for p in reversed(sq_annihilators):
-                if( basis_I.get_bit(p) == 0):
-                    destroyed=True
-                    break
-
-                basis_I.set_bit(p, 0)
-
-            # then over creators
-            for p in reversed(sq_creators):
-                if (basis_I.get_bit(p) == 1):
-                    destroyed=True
-                    break
-
-                basis_I.set_bit(p, 1)
-
-            if not destroyed:
-
-                I = basis_I.add()
 
-                ## check for correct dets
-                det_I = integer_to_ref(I, self._nqb)
-                nel_I = sum(det_I)
-                cor_spin_I = correct_spin(det_I, 0)
+            qc_temp = qforte.Computer(self._nqb)
+            qc_temp.apply_circuit(self._Uprep)
+            qc_temp.apply_operator(sq_op.jw_transform())
+            sign_adjust = qc_temp.get_coeff_vec()[self._reversed_excitation_dictionary[m]]
```

Comment:
Why, exactly is `sign_adjust` needed? That isn't clear to me from the original code.

Reply:
I had the same question as you. In fact, when I first saw that I tried to remove it. The numerical results did not make sense. At some point I believe I figured out why it was needed, but now this knowledge has evaporated.

Reply:
I'll try to figure it out myself.

Reply:
Got it.
coeffs_I is U^HU|Phi>. We want to measure <Phi| K U^HU|Phi>. We know which vector element of coeffs_I to probe, but the corresponding basis element might not have the same parity as <Phi |K. We correct for that by checking the parity of the corresponding basis element in K|Phi>.
If that makes sense, this _really_ should be a comment.

Reply:
Yep. Fermionic antisymmetry. I will add a comment.